addons/sale/views/sale_views.xml

```diff
@@ -661,7 +661,7 @@
                         <page string="Other Info" name="other_information">
                             <group>
                                 <group name="sales_person" string="Sales">
-                                    <field name="user_id" domain="[('share', '=', False)]" widget="many2one_avatar_user"/>
+                                    <field name="user_id" domain="[('share', '=', False),('company_id','=',company_id)]" widget="many2one_avatar_user"/>
```

Comment:
I think this domain should be removed in the view and correctly integrated in the `domain` specified on the `user_id` field instead (because atm, we also show non salesman users, which is wrong as well).